### PR TITLE
Document creating custom checks

### LINKED ISSUE #1210: Document writing a custom Check using a task.
Here is an example:

```
        import "strings"
        import "regexp"
        import "influxdata/influxdb/monitor"
        import "influxdata/influxdb/v1"
        task_data = from(bucket: "_tasks")
            |> range(start: -1m, stop: now())
            |> filter(fn: (r) => r["_measurement"] == "runs")
            |> filter(fn: (r) => r["_field"] == "logs")
            |> map(fn: (r) => ({ r with name: regexp.findString(r: /[0-9]+\. downsample_[a-z_]+_[0-9]+[mhd]/, v: r._value) }))
            |> map(fn: (r) => ({ r with every: regexp.findString(r: /every: ([0-9]+[smhd])+/, v: r._value) }))
            |> map(fn: (r) => ({ r with every: if r.every != "" then strings.split(v: r.every , t: " ")[1] else "" }))
            |> filter(fn: (r) => r.name != "")
            |> drop(columns: ["_value","_start","_stop"])
            |> group(columns: ["name","taskID","status","_measurement"])
            |> map(fn: (r) => ({ r with _value: if r.status == "failed" then 1 else 0 }))
            |> last()
        check = {
            _check_id: "00000000deadbeef",  // can be anything [0-9a-f]{16}
            _check_name: "00. Downsample Monitoring Task", // Can be anything
            _type: "threshold", // custom is probably better here
            tags: {}, // anything you might enter in the custom tags section of the UI when creating a check
        }
        ok = (r) =>
            (r["logs"] == 0)
        crit = (r) =>
            (r["logs"] == 1)
       // info and warn are also options here
       // can be any expression
       // assume the expression is executed after a pivot, which is why we use `r["logs"]` instead of `r["_value"]`

        messageFn = (r) =>
            ("The downsampling task: ${r.taskID} - ${r.name} has a status of ${r.status}")
        // The status message for the check

        // data must have _field, _measurement when pivoting
        task_data
            |> v1["fieldsAsCols"]()
            |> monitor["check"](
                data: check,
                messageFn: messageFn,
                ok: ok,
                crit: crit,
                //warn: warn,
                //info: info,
            )
```

### PR DESCRIPTION
**Pull request by pierwill**
Closes #1210.

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+1. In the InfluxDB UI, select **Tasks** in the navigation menu on the left.
+
+    {{< nav-icon "tasks" >}}
+
+2. Click **{{< icon "plus" >}} Create Task**, and then select **New Task**.
+3. In the **Name** field, enter a descriptive name, for example, **Alert on task failure**,
+   and then enter how often to run the task in the **Every** field (for example, `10m`).
+   For more detail, such as using cron syntax or including an offset, see [Task configuration options](/influxdb/v2.0/process-data/task-options/).
+4. Enter the [example script](#example-alert-task-script) and click **Save**.
```
[the remark this answers is not in the document]
Comment (pierwill) [reply content/influxdb/v2.0/monitor-alert/custom-checks.md:35]: Makes sense to me!

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+---
```
[the remark this answers is not in the document]
Comment (pierwill) [reply content/influxdb/v2.0/monitor-alert/custom-checks.md:1]: Would we also move the email notification doc there?

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+3. In the **Name** field, enter a descriptive name,
+   and then enter how often to run the task in the **Every** field (for example, `10m`).
+   For more detail, such as using cron syntax or including an offset, see [Task configuration options](/influxdb/v2.0/process-data/task-options/).
+4. Enter the Flux script for your custom check, including the [`monitor.check`](/influxdb/v2.0/reference/flux/stdlib/monitor/check/) function.
+
+### Example: Monitor failed tasks
+
+Receive an alert when a task fails.
+
+The script below is fairly complex, and can be used an a framework for similar tasks.
```
Comment (kelseiv) [inline content/influxdb/v2.0/monitor-alert/custom-checks.md:38]: as an

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	|> drop(columns: ["_value", "_start", "_stop"])
+	|> group(columns: ["name", "taskID", "status", "_measurement"])
+	|> map(fn: (r) =>
+		({r with _value: if r.status == "failed" then 1 else 0}))
+	|> last()
+check = {
+	_check_id: "0000000000000001",
+	_check_name: "Failed Tasks Check",
+	_type: "custom",
+	tags: {},
```
Comment (kelseiv) [inline content/influxdb/v2.0/monitor-alert/custom-checks.md:74]: Perhaps we could add comments (or clarify on line 44):
- check id: "16 alphanumeric characters (letters must be lowercase)"
- check name: "string"
- type: point out here you could also use deadman or threshold

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+menu:
+  influxdb_2_0:
+    parent: Monitor & alert
+weight: 201
+v2.0/tags: [alerts, checks, tasks, Flux]
+---
+
+In the UI, you can create two kinds of [checks](/influxdb/v2.0/reference/glossary/#check):
+[`Threshold check`](/influxdb/v2.0/monitor-alert/checks/create/#threshold-check) and
+[`Deadman check`](/influxdb/v2.0/monitor-alert/checks/create/#deadman-check).
```
Comment (kelseiv) [inline content/influxdb/v2.0/monitor-alert/custom-checks.md:15]: to visually tie w line 19, we could format like this:
[`threshold` check](/influxdb/v2.0/monitor-alert/checks/create/#threshold-check) and
[`deadman` check](/influxdb/v2.0/monitor-alert/checks/create/#deadman-check).

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+Using a Flux task, you can create a custom check that provides a couple advantages:
+
+- Set up custom criteria for your alert (other than `threshold` and `deadman`).
```
Comment (russorat) [inline content/influxdb/v2.0/monitor-alert/custom-checks.md:19]:
```suggestion
- Customize and transform the data you would like to use for the check.
- Set up custom criteria for your alert (other than `threshold` and `deadman`).
```

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+- Share custom checks across distributed instances.
```
Comment (russorat) [inline content/influxdb/v2.0/monitor-alert/custom-checks.md:20]: I'm not sure i understand this one. you can also share standard checks this way i believe.

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+2. Click **{{< icon "plus" >}} Create Task**, and then select **New Task**.
+3. In the **Name** field, enter a descriptive name,
+   and then enter how often to run the task in the **Every** field (for example, `10m`).
+   For more detail, such as using cron syntax or including an offset, see [Task configuration options](/influxdb/v2.0/process-data/task-options/).
+4. Enter the Flux script for your custom check, including the [`monitor.check`](/influxdb/v2.0/reference/flux/stdlib/monitor/check/) function.
+
+### Example: Monitor failed tasks
+
+Receive an alert when a task fails.
```
Comment (russorat) [inline content/influxdb/v2.0/monitor-alert/custom-checks.md:36]: Technically, this script doesn't send the alert, it only "checks" for failed tasks. You would need to combine this with a notification rule in order to get something sent to you in slack for example.

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+1. In the InfluxDB UI, select **Tasks** in the navigation menu on the left.
+
+    {{< nav-icon "tasks" >}}
+
+2. Click **{{< icon "plus" >}} Create Task**, and then select **New Task**.
+3. In the **Name** field, enter a descriptive name,
+   and then enter how often to run the task in the **Every** field (for example, `10m`).
+   For more detail, such as using cron syntax or including an offset, see [Task configuration options](/influxdb/v2.0/process-data/task-options/).
+4. Enter the Flux script for your custom check, including the [`monitor.check`](/influxdb/v2.0/reference/flux/stdlib/monitor/check/) function.
+
```
Comment (russorat) [inline content/influxdb/v2.0/monitor-alert/custom-checks.md:33]: Maybe you can mention in a note somewhere that you can see the flux code generated by a check you've built in the UI using the api endpoint /checks/checkID/query

File: content/influxdb/v2.0/monitor-alert/custom-checks.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+```js
+import "strings"
+import "regexp"
+import "influxdata/influxdb/monitor"
+import "influxdata/influxdb/v1"
+
+option task = {name: "Failed Tasks Check", every: 1h, offset: 4m}
+
+task_data = from(bucket: "_tasks")
+	|> range(start: -1h, stop: now())
```
Comment (russorat) [inline content/influxdb/v2.0/monitor-alert/custom-checks.md:59]: you can use `-task.every` here.

**Pull request merged**